app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt

```diff
@@ -0,0 +1,350 @@
+      val segmentAngle = 360f / segments.size
+      var requiredRotation = ((targetSegment - currentSegment) * segmentAngle)
+
+      // Calculate where we'll land within the target segment
+      val landingAngle = (270 - ((rotation - requiredRotation) % 360)) % 360
+      val angleWithinSegment = landingAngle % segmentAngle
+
+      val shouldTriggerNearMiss = Math.random() < 0.7f // 70% chance for near-miss
+      var randomOffset = 0f
+
+      when (targetSegment) {
+        0 -> { // Nothing segment
+          if (shouldTriggerNearMiss) {
+            // For Nothing, we want to get close to the right edge (near Legendary)
+            val maxOffsetRight = segmentAngle - angleWithinSegment
+            // Use 90-95% of the distance to get very close to Legendary
+            randomOffset = maxOffsetRight * (0.90f + (Math.random() * 0.05f).toFloat())
+            Log.d(TAG, "Triggering near-miss on Nothing segment (close to Legendary)")
+          } else {
+            // Normal random offset within safe bounds
+            val maxOffsetLeft = angleWithinSegment
+            val maxOffsetRight = segmentAngle - angleWithinSegment
+            val safeOffset = minOf(maxOffsetLeft, maxOffsetRight) * 0.9f
+            randomOffset = (Math.random() * 2 * safeOffset - safeOffset).toFloat()
+          }
+        }
+        2 -> { // Common segment
+          if (shouldTriggerNearMiss) {
+            // For Common, we want to get close to the left edge (near Legendary)
+            val maxOffsetLeft = angleWithinSegment
+            // Use only 5-10% of the distance to get very close to Legendary
+            randomOffset = -maxOffsetLeft * (0.90f + (Math.random() * 0.05f).toFloat())
+            Log.d(TAG, "Triggering near-miss on Common segment (close to Legendary)")
+          } else {
+            // Normal random offset within safe bounds
+            val maxOffsetLeft = angleWithinSegment
+            val maxOffsetRight = segmentAngle - angleWithinSegment
+            val safeOffset = minOf(maxOffsetLeft, maxOffsetRight) * 0.9f
+            randomOffset = (Math.random() * 2 * safeOffset - safeOffset).toFloat()
+          }
+        }
+        else -> {
+          // For other segments, use normal random offset within safe bounds
+          val maxOffsetLeft = angleWithinSegment
+          val maxOffsetRight = segmentAngle - angleWithinSegment
+          val safeOffset = minOf(maxOffsetLeft, maxOffsetRight) * 0.9f
+          randomOffset = (Math.random() * 2 * safeOffset - safeOffset).toFloat()
+        }
+      }
+
+      requiredRotation += randomOffset
+
+      // Ensure we're always rotating clockwise
+      if (requiredRotation <= 0) requiredRotation += 360
+
+      // Add 6-7 full rotations for effect
+      val fullRotations = (6..7).random() * 360
+      targetRotation = rotation - (requiredRotation + fullRotations)
+
+      Log.d(TAG, "Landing angle within segment: $angleWithinSegment")
```

Comment:
I think there may be a little too many tags. I know they're meant for logging but there's so many that it's basically impossible to keep track of them all which harms the feature